libs/forms/core/src/form-tree-node.service.ts

```diff
@@ -28,35 +33,92 @@ export class DynFormTreeNode<TControl extends AbstractControl = FormGroup>{
   get control(): TControl {
     return this._control;
   }
+  get isRoot(): boolean {
+    return this.isolated || !this.parent;
+  }
 
   // control.path relative to the root
   get path(): string[] {
     return [
-      ...(!this.isolated ? this.parent?.path ?? [] : []),
+      ...(!this.isRoot ? this.parent.path : []),
       this._name ?? '',
     ].filter(Boolean);
   }
 
   private _name?: string;
   private _instance!: DynInstanceType;
   private _control!: TControl;
+  private _matchers?: DynControlMatch[];
+
+  private _unsubscribe = new Subject<void>();
 
   constructor(
     private readonly formFactory: DynFormFactory,
+    private readonly formMatchers: DynFormMatchers,
     private readonly logger: DynLogger,
     // parent node should be set for all except the root
     @Optional() @SkipSelf() public readonly parent: DynFormTreeNode,
   ) {}
 
-  setControl(control: TControl, instance = DynInstanceType.Group): void {
-    this.logger.nodeControl();
+  /**
+   * Feature methods
+   */
 
-    // manual setup with no wiring nor config validation
-    this._instance = instance;
-    this._control = control;
+  // let the TreeNode know of an incoming hook
+  callHook(event: DynControlHook): void {
+    this.logger.hookCalled(event.hook, this.path, event.payload);
+
+    this.hook$.next(event);
+  }
+
+  // query for a control upper in the tree
+  query(path: string, searchNodes = false): AbstractControl|null {
+    /* eslint-disable @typescript-eslint/no-this-alias */
+    let node: DynFormTreeNode<AbstractControl> = this;
+    let result: AbstractControl|null;
+
+    do {
+      // query by form.control and by node.path
+      result = node.control.get(path)
+        ?? (searchNodes ? node.select(path) : null)
+        ?? null;
+      // move upper in the tree
+      node = node.parent;
+    } while (!result && node);
+
+    return result;
   }
```

Comment:
@Yumitoya8569 looks familiar? :)
perhaps here `query` searchs in ancestors and `select` in the children

Reply:
@matheo 
Children selector bring little benefit
```
{
   a:{
      a1:''
   },
   b:{
     b1:''
   }
}

// ancestors query
b1.query('a.a1');

// select children (benefit: `path` is shorter)
b1.select('a1');
```
but user need more careful when use it.
```
{
   a:{
      opt:''
   },
   b:{
     opt:''
   }
}

// select children (users should be careful what they are doing)
b.select('opt');
```

Reply:
`query` will stop on the first coincidence (as it depends on `select`)
and `b.query('opt', true);` won't return a control in that structure, because `a` doesn't have a `opt` child.
I will see how to add `RegExp` support in the paths, but still it will stop in the first coincidence.

Reply:
@matheo 
Sorry, I didn't notice that 'query.(path, true)' won't search all the descendants node.

But more confused now. Any use case for `query.(path, false)` and `query.(path, true)` to show their diff ?
THX

Reply:
@Yumitoya8569 the `searchNodes: boolean` is useful to go inside FormArrays basically
I've had forms with arrays where you're interested in the first item (most recent employment, etc)
so you can specify a path like `employments.0.paymentFrequency`
in this case `FormGroup.get` is not able to fetch this field, but `query('employments.0.paymentFrequency', true)` is able :slightly_smiling_face: 

going back to your structure:
```
{
   a:{
      a1:{  opt:'' },
      a2:{  opt:'' }

   },
   b:''
}
```
the path to query a specific field needs to match the parent complete path, like `a.a1.opt` or `a.a2.opt`.

Reply:
@matheo 
`FormGroup.get` should work. Am i missing something? 
https://stackblitz.com/edit/angular-ivy-dngl7o?file=src/app/app.component.ts

Reply:
oh weird, it didn't retrieved the control while I was testing :thinking:
I will check if it was a matter of the Form bootstrapping, thank you!